go/libraries/doltcore/schema/alterschema/addpk.go

```diff
@@ -52,10 +52,11 @@ func AddPrimaryKeyToTable(ctx context.Context, table *doltdb.Table, tableName st
 	for ord, col := range sch.GetAllCols().GetColumns() {
 		if i, ok := pkColOrdering[col.Name]; ok {
 			pkOrdinals[i] = ord
-			col.IsPartOfPK = true
+			// Can't assign IsPartOfPK true, before checking if IsNullable, since it'll always be false
```

Comment:
good catch here, I would go a step further and add a helper method like `addSingleNotNullConstraint()` that adds not null constraints iff it doesn't exist, and then validates that only one exists. 
Schema modification operations have proven to be very fragile, so I think we should be extra explicit when we make them and add extra validation as backup